compiler/rustc_codegen_llvm/src/debuginfo/di_builder.rs

```diff
@@ -0,0 +1,115 @@
+//! Safe wrappers for [`DIBuilder`] FFI functions.
+
+use libc::c_uint;
+use rustc_abi::{Align, Size};
+
+use crate::llvm::debuginfo::{DIBuilder, DIFlags};
+use crate::llvm::{self, Metadata};
+
+impl<'ll> DIBuilder<'ll> {
+    pub(crate) fn create_subroutine_type(
+        &self,
+        parameter_types: &[Option<&'ll Metadata>],
+        flags: DIFlags,
+    ) -> &'ll Metadata {
+        unsafe {
+            llvm::LLVMDIBuilderCreateSubroutineType(
+                self,
```

Comment:
Okay, so, it's somewhat hard to audit the safe wrappers if I don't know what safety constraints they are filling. I can guess, but I shouldn't have to. Either there should be safety comments on the wrapper decls or they should be on this block, so that future additions to these functions don't fuck up and change the parts of the code that have to be a certain way.

Reply:
like it could be, idk, instead of exhaustively listing param relationships
```rust
// SAFETY: this call is safe if we know the DIBuilder is live, which we do because
// we have it by-ref, and the rest is just translating slices to two params
```